### PR TITLE
Add <ArticleFooter/>

### PR DESCRIPTION
**Pull request by hueitan**
more to be done

- [x] i18n footer content
- [x] footer title in toc
- [x] thumbnail selected styling
- [x] highlight first item when the page load (article link)

### CODE REVIEW

File: src/components/QuickFacts.js [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+  return (
+    <Fragment>
+      <div
+        class='quickfacts'
+        ref={containerRef}
+        dangerouslySetInnerHTML={{ __html: article.infobox }}
+      />
+    </Fragment>
+  )
+}
```
Comment (hueitan) [inline src/components/QuickFacts.js]: ahhh! that windows LF CRLF issue!

**Comment by hueitan** [comment]
This feature is ready to be reviewed now

but will need to consult the design of the suggested articles placeholder, will do it in the seperate task.

File: i18n/en.json [hunk 2]
```diff
@@ -40,5 +42,7 @@
     "settings-term": "Terms of use",
     "settings-rate": "Rate the app",
     "settings-help-feedback": "Help and feedback",
-    "settings-about-app": "About the app"
+    "settings-about-app": "About the app",
+    "toc-footer": "Footer",
```
Comment (stephanebisson) [inline i18n/en.json]: I don't think this should be "Footer". I've asked on phab.

File: i18n/en.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "app-title": "Wikipedia",
     "app-subtitle": "The free encyclopedia",
     "app-description": "With the free Wikipedia app, search and explore information from all over the world in 300+ languages.",
+    "content-license": "Content is available under $1 unless otherwise noted.",
```
Comment (stephanebisson) [inline i18n/en.json]: Why is the license a parameter here?

File: src/components/ArticleFooter.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+              </div>
+              { item.thumbnail && <div class='img'><img src={item.thumbnail.source} /></div> }
+            </a>
+          })}
+        </div>
+        <h2>
+          <img src='/images/wikipedia-wordmark-en.png' height='18' width='116' />
+        </h2>
+        <p class='license' dangerouslySetInnerHTML={
+          { __html: i18n.i18n('content-license', '<a class=\'external\' rel=\'mw:ExtLink\' href=\'https://creativecommons.org/licenses/by-sa/3.0/\'>CC BY-SA 3.0</a>') }
```
Comment (stephanebisson) [inline src/components/ArticleFooter.js]: Building this link externally makes it untranslatable. In this case the link should be included in the message. 

See https://github.com/wikimedia/mediawiki-extensions-WikimediaMessages/blob/master/i18n/wikimedia/en.json#L217

File: src/components/ArticleFooter.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        />
+        <p class='browser'>
+          <a class='external' rel='mw:ExtLink' href={`https://${lang}.m.wikipedia.org/w/index.php?title=${title}`}>
```
Comment (stephanebisson) [inline src/components/ArticleFooter.js]: Should we encode `title` in the URL?

File: src/components/QuickFacts.js (hunk 1 above)
Comment (stephanebisson) [reply src/components/QuickFacts.js]: Are you really using Windows?

File: src/api/getArticle.js [hunk 6]
```diff
@@ -72,6 +74,15 @@ export const getArticle = (lang, title) => {
       }
     })
 
+    // footer
+    sections.push({
+      title: i18n.i18n('toc-footer'),
```
Comment (stephanebisson) [inline src/api/getArticle.js]: You can't use i18n here. The API output is transformed and cached independent of the language.

Comment (hueitan) [reply src/api/getArticle.js]: true, and I think we have another problem here, we can change the article language, but we didn't fetch the language file for that language, is that fine the footer here "suggested articles" always the app language translation?

File: src/components/ArticleFooter.js (hunk 5 above)
Comment (hueitan) [reply src/components/ArticleFooter.js]: I'm thinking of adding `utils.buildWpMobileWeburl( lang, title )` here and encoding within the util

so we don't write the link here

**Comment by hueitan** [comment]
Ready to be reviewed now, `Edit history` will be made in another pr.

File: src/components/ArticlePreview.js [hunk 7]
```diff
@@ -1,36 +1,36 @@
...
-          <div class='title' dangerouslySetInnerHTML={{ __html: data.title }} />
-          <div class='description' dangerouslySetInnerHTML={{ __html: data.description }} />
-        </div>
-        { data.imageUrl && <div class='img'><img src={data.imageUrl} /></div> }
-      </div>
-      <div class='preview-text' dangerouslySetInnerHTML={{ __html: data.preview }} />
-    </div>
-  )
-}
+import { h } from 'preact'
```
[the remark this answers is not in the document]
Comment (hueitan) [reply src/components/ArticlePreview.js]: truee ... Windows stuff, I will figure it out.

**Comment by hueitan** [comment]
Close and reopen a new pr.

**Pull request closed without merging**